From f788178536afa2e40d577d544b3f942a0476b2d3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=A5=A0=E3=81=A1=E3=82=83=E3=82=93?= <kusu@kusu.moe>
Date: Fri, 5 Dec 2025 21:05:54 +0800
Subject: [PATCH] Patch sysctl.c Table to Prevent Register Failed

---
 kernel/sysctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sysctl.c b/kernel/sysctl.c
index 94ffe9e01815..04a927a5d753 100644
--- a/kernel/sysctl.c
+++ b/kernel/sysctl.c
@@ -1519,7 +1519,7 @@ static const struct ctl_table sysctl_subsys_table[] = {
 #endif
 #ifdef CONFIG_SCHED_ALT
 	{
-		.procname	= "yield_type",
+		.procname	= "custom_yield_type",
 		.data		= &sched_yield_type,
 		.maxlen		= sizeof (int),
 		.mode		= 0644,
-- 
2.51.0

