From 9e99500621ecf2d7af9103ab7fab97f627eaf1bb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=A5=A0=E3=81=A1=E3=82=83=E3=82=93?= <kusu@kusu.moe>
Date: Sat, 15 Nov 2025 19:51:06 +0800
Subject: [PATCH] Update syscalls.c

---
 kernel/sched/syscalls.c | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/kernel/sched/syscalls.c b/kernel/sched/syscalls.c
index 9cd6a7208cc7..b36416fdbd50 100644
--- a/kernel/sched/syscalls.c
+++ b/kernel/sched/syscalls.c
@@ -1593,7 +1593,26 @@ static void do_sched_yield(void)
 {
 	struct rq_flags rf;
 	struct rq *rq;
+#ifdef CONFIG_SCHED_ALT
+    struct task_struct *p;
+
+    if (!sched_yield_type)
+        return;
+
+    rq = this_rq_lock_irq(&rf);
+
+    schedstat_inc(rq->yld_count);
 
+    p = current;
+    if (rt_task(p)) {
+        if (task_on_rq_queued(p))
+            requeue_task(p, rq);
+    } else if (rq->nr_running > 1) {
+        do_sched_yield_type_1(p, rq);
+        if (task_on_rq_queued(p))
+            requeue_task(p, rq);
+    }
+#else /* !CONFIG_SCHED_ALT */
 	if (!sysctl_sched_yield_type)
 		return;
 
@@ -1603,7 +1622,7 @@ static void do_sched_yield(void)
 
 	if (sysctl_sched_yield_type > 1)
 		current->sched_class->yield_task(rq);
-
+#endif /* !CONFIG_SCHED_ALT */
 	preempt_disable();
 	rq_unlock_irq(rq, &rf);
 	sched_preempt_enable_no_resched();
-- 
2.51.0

